block/provider/delete_snapshot.go

```diff
@@ -29,16 +29,17 @@ import (
 
 // DeleteSnapshot delete snapshot
 func (vpcs *VPCSession) DeleteSnapshot(snapshot *provider.Snapshot) error {
-	vpcs.Logger.Info("Entry DeleteSnapshot", zap.Reflect("snapshotID", snapshot.SnapshotID))
-	defer vpcs.Logger.Info("Exit DeleteSnapshot", zap.Reflect("snapshotID", snapshot.SnapshotID))
-	defer metrics.UpdateDurationFromStart(vpcs.Logger, "DeleteSnapshot", time.Now())
+	vpcs.Logger.Info("Entry DeleteSnapshot", zap.Reflect("snapshot", snapshot))
 
 	var err error
 	if snapshot == nil {
 		err = userError.GetUserError("InvalidSnapshotID", nil, nil)
 		return err
 	}
 
+	defer vpcs.Logger.Info("Exit DeleteSnapshot", zap.Reflect("snapshotID", snapshot.SnapshotID))
+	defer metrics.UpdateDurationFromStart(vpcs.Logger, "DeleteSnapshot", time.Now())
```

Comment:
@sameshai can you please add details to the description on why these code changes are needed.